File: Programmers/level1/divisorCntAndAdd/divisorCntAndAdd.py
# 문제 출처 : https://programmers.co.kr/learn/courses/30/lessons/77884

def solution(left, right):
    answer = 0
    for i in range(left, right+1):
        cnt = 0
        for j in range(1, i+1):
            if i % j == 0:
                cnt += 1
        if cnt % 2 == 0:
            answer += i
        else:
            answer -= i
    return answer

File: Programmers/level1/divisorCntAndAdd/divisorCntAndAdd.js
// https://programmers.co.kr/learn/courses/30/lessons/77884?language=javascript

export const solution = (left, right) => {
    let answer = 0;
    for(let j = left; j < right+1; j++){
        let num = divisorCnt(j);
        if(num % 2 === 0){
            answer += j;
        } else {
            answer -= j;
        }
    }
    return answer;
}

const divisorCnt = (num) => {
    let cnt = 0
    for(let i = 1; i < num+1; i++){
        if(num % i === 0){
            cnt += 1;
        }
    }
    return cnt
}
